game-core/src/main/java/games/strategy/triplea/ui/DicePanel.java

```diff
@@ -27,59 +27,80 @@
   public DicePanel(final UiContext uiContext, final GameData data) {
     this.uiContext = uiContext;
     this.data = data;
-    setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+    setLayout(new GridBagLayout());
+    setBorder(BorderFactory.createEmptyBorder(8, 16, 0, 0));
   }
 
   void setDiceRollForBombing(final List<Die> dice, final int cost) {
     removeAll();
+    add(new JLabel("Cost: " + cost, SwingConstants.LEFT));
     add(create(dice));
-    add(Box.createVerticalGlue());
-    add(new JLabel("Cost:" + cost));
-    invalidate();
+    addBottomLabel(new JLabel());
```

Comment:
Any objection to having the bottom label be returned from the function? As is, the method `addBottomLabel` relies on side effects. To understand what is happening, you have to dive into the method, which lacks abstraction. We also have some inconsistency from the other methods that add returned values compared to this last one that adds a component as a side-effect.

Reply:
Uhm, it's not a side effect, it's the main purpose of the function?

Reply:
Oh, do you mean the setVerticalAlignment() call?

Given that we no longer use a label with content (before, we used to for total hits) and this is just for taking up the remaining vertical space, we can simplify it. Done.